File: src/com/mycompany/myapp/entities/LigneFacture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author Pc Anis
 */
public class LigneFacture {
    private int id_ligne_fac;
    private Facture id_facture;
    private Livre id_livre;
    private int id_user;
    private float mnt;
    private int qte;

    public LigneFacture() {
    }

    public LigneFacture(int id_ligne_fac, Facture id_facture, Livre id_livre, int id_user, float mnt, int qte) {
        this.id_ligne_fac = id_ligne_fac;
        this.id_facture = id_facture;
        this.id_livre = id_livre;
        this.id_user = id_user;
        this.mnt = mnt;
        this.qte = qte;
    }

    public LigneFacture(Facture id_facture, Livre id_livre, int id_user, float mnt, int qte) {
        this.id_facture = id_facture;
        this.id_livre = id_livre;
        this.id_user = id_user;
        this.mnt = mnt;
        this.qte = qte;
    }

    public int getId_ligne_fac() {
        return id_ligne_fac;
    }

    public void setId_ligne_fac(int id_ligne_fac) {
        this.id_ligne_fac = id_ligne_fac;
    }

    public Facture getId_facture() {
        return id_facture;
    }

    public void setId_facture(Facture id_facture) {
        this.id_facture = id_facture;
    }

    public Livre getId_livre() {
        return id_livre;
    }

    public void setId_livre(Livre id_livre) {
        this.id_livre = id_livre;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    public float getMnt() {
        return mnt;
    }

    public void setMnt(float mnt) {
        this.mnt = mnt;
    }

    public int getQte() {
        return qte;
    }

    public void setQte(int qte) {
        this.qte = qte;
    }

    @Override
    public String toString() {
        return "LigneFacture{" + "id_ligne_fac=" + id_ligne_fac + ", id_facture=" + id_facture + ", id_livre=" + id_livre + ", id_user=" + id_user + ", mnt=" + mnt + ", qte=" + qte + '}';
    }
    
    
    
}

File: src/Entity/LigneFacture.php
<?php

namespace App\Entity;
use App\Repository\LigneFactureRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: LigneFactureRepository::class)]
class LigneFacture
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idLigneFac=null;

    #[ORM\Column]
    private ?float $mnt=null;

    #[ORM\Column]
    private ?int $qte=null;

    
    #[ORM\ManyToOne(targetEntity: Utilisateur::class, inversedBy: 'lignefactures')]
    #[ORM\JoinColumn(name: 'id_user', referencedColumnName: 'id_user')]
    private ?Utilisateur $idUser=null;

    
    #[ORM\ManyToOne(targetEntity: Facture::class, inversedBy: 'lignefactures')]
    #[ORM\JoinColumn(name: 'id_facture', referencedColumnName: 'id_facture')]
    private ?Facture $idFacture=null;

  
    #[ORM\ManyToOne(targetEntity: Livre::class, inversedBy: 'lignefactures')]
    #[ORM\JoinColumn(name: 'id_livre', referencedColumnName: 'id_livre')]
    private ?Livre $idLivre=null;

    public function getIdLigneFac(): ?int 
    {
        return $this->idLigneFac;
    }

    public function getMnt(): ?float
    {
        return $this->mnt;
    }

    public function setMnt(float $mnt): self
    {
        $this->mnt = $mnt;

        return $this;
    }

    public function getQte(): ?int
    {
        return $this->qte;
    }

    public function setQte(int $qte): self
    {
        $this->qte = $qte;

        return $this;
    }

    public function getIdUser(): ?Utilisateur
    {
        return $this->idUser;
    }

    public function setIdUser(?Utilisateur $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }

    public function getIdFacture(): ?Facture
    {
        return $this->idFacture;
    }

    public function setIdFacture(?Facture $idFacture): self
    {
        $this->idFacture = $idFacture;

        return $this;
    }

    public function getIdLivre(): ?Livre
    {
        return $this->idLivre;
    }

    public function setIdLivre(?Livre $idLivre): self
    {
        $this->idLivre = $idLivre;

        return $this;
    }


}
